### DISCUSSION TITLE
How to make Milvus use password login when starting docker-compose?

### DISCUSSION THREAD

**Discussion by sunshineflg** [milvus-io/milvus#25974] [Q&A and General discussion]
docker-compose 启动时怎么让milvus使用密码登录？milvus的容器没有编辑器，apt install vim   E: Unable to locate package vim

**Comment by yhmo** [comment]
milvus启动时的配置文件放在容器里的 /milvus/configs/milvus.yaml 

要改配置，必须改这个文件。

为了改这个文件，可以把这个文件路径通过docker的命令映射到外部路径，这里讲了方法：https://milvus.io/docs/configure-docker.md
主要就是在docker-compose.yml里加那么一句：
```
  standalone:
    ......
    volumes:
      - [外部milvus.yaml所在的路径]/milvus.yaml:/milvus/configs/milvus.yaml
```

把milvus.yaml映射到外部路径后，修改外部路径那个milvus.yaml，设置 common.security.authorizationEnabled =true，这里讲了：[Set common.security.authorizationEnabled in milvus.yaml](https://milvus.io/docs/authenticate.md)

改好之后docker-compose up -d启动

**Comment by yhmo** [comment]
另外：milvus社区里鼓励使用英文，尽量使用英文提问和对话，谢谢

**Reply by mmmmmmnnmmmmmm** [reply]
I have set in the configuration file, how can I modify the default root password during ?authorizationEnabled: truedocker compose up -d

security:
  authorizationEnabled: true
  superUsers:
  defaultRootPassword: 85uHp7kHKPUE48NDakvZ  # This password is for testing
  rootShouldBindRole: false
  enablePublicPrivilege: true


However, I encountered an error when trying to log in:

milvus_cli > connect -uri http://localhost:19531 -t root:85uHp7kHKPUE48NDakvZ
2025-12-08 12:58:23,985 [ERROR][handler]: grpc RpcError: [__internal_register], <_InactiveRpcError: StatusCode.UNAUTHENTICATED, auth check failure, please check username and password are correct>, <Time:{'RPC start': '2025-12-08 12:58:23.982696', 'gRPC error': '2025-12-08 12:58:23.985483'}> (decorators.py:151)
Connect to Milvus error!<_InactiveRpcError of RPC that terminated with:
    status = StatusCode.UNAUTHENTICATED
    details = "auth check failure, please check username and password are correct"
    debug_error_string = "UNKNOWN:Error received from peer  {created_time:"2025-12-08T12:58:23.985290912+08:00", grpc_status:16, grpc_message:"auth check failure, please check username and password are correct"}"
>


Could anyone help me solve this authentication issue or clarify the correct approach to modifying the root password during the Docker container startup?

**Reply by yhmo** [reply]
Unzip this file to your local:
[milvus.zip](https://github.com/user-attachments/files/24025080/milvus.zip)

Stop all milvus instance. cmd into the unzip diorectory, then "docker compose up -d" to start the standalone milvus.
Use this script to connect it:
```
from pymilvus import (
    MilvusClient
)

client = MilvusClient(uri="http://localhost:19530",
                      user="root",
                      password="85uHp7kHKPUE48NDakvZ")
print("connected, server version: ", client.get_server_version())
```
It works well.

There are two files in the unzip directory: docker-compose.yml and milvus.yaml.
Open the docker-compose.yml from the unzip directory, we can see the milvus.yaml is mapping to the container inside:
```
standalone:
    ....
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/milvus.yaml:/milvus/configs/milvus.yaml
```
Open the milvus.yaml from the unzip directory, we can see the authorizationEnabled is true and defaultRootPassword is 85uHp7kHKPUE48NDakvZ.
```
common:
  security:
    authorizationEnabled: true
    defaultRootPassword: 85uHp7kHKPUE48NDakvZ
```

**Reply by mmmmmmnnmmmmmm** [reply]
The issue has been resolved. I changed the password to a purely alphabetic one.

**Reply by mmmmmmnnmmmmmm** [reply]
> 将此文件解压到本地：[milvus.zip](https://github.com/user-attachments/files/24025080/milvus.zip)
> 
> 停止所有 Milvus 实例。cmd 进入解压指令，然后“docker compose up -d”启动独立 milvus。使用这个脚本连接它：
> 
> ```
> from pymilvus import (
>     MilvusClient
> )
> 
> client = MilvusClient(uri="http://localhost:19530",
>                       user="root",
>                       password="85uHp7kHKPUE48NDakvZ")
> print("connected, server version: ", client.get_server_version())
> ```
> 
> 效果很好。
> 
> 解压目录中有两个文件：docker-compose.yml 和 milvus.yaml。从解压目录打开docker-compose.yml，我们可以看到 milvus.yaml 映射到内部容器：
> 
> ```
> standalone:
>     ....
>     volumes:
>       - ${DOCKER_VOLUME_DIRECTORY:-.}/milvus.yaml:/milvus/configs/milvus.yaml
> ```
> 
> 从解压目录打开milvus.yaml，我们可以看到authorizationEnabled为真，defaultRootPassword是85uHp7kHKPUE48NDakvZ。
> 
> ```
> common:
>   security:
>     authorizationEnabled: true
>     defaultRootPassword: 85uHp7kHKPUE48NDakvZ
> ```

Received, thank you for your method, I will try it later. Thank you so much for your patient reply!
